File: app/src/main/java/com/example/webtoonapp/bannerAdapter.java
package com.example.webtoonapp;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.constraintlayout.utils.widget.ImageFilterView;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager2.widget.ViewPager2;

import java.util.ArrayList;

public class bannerAdapter extends PagerAdapter {

    private Context mContext;
    private ArrayList<Integer> bannerImage;

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        LayoutInflater inflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View view = inflater.inflate(R.layout.fragment_main, null);

        ViewPager2 viewPager2 = view.findViewById(R.id.vp2_banner);
        viewPager2.setBackgroundResource(bannerImage.get(position));

        container.addView(view);

        return view;
    }

    @Override
    public int getCount() {
        return bannerImage.size();
    }

    public bannerAdapter(Context context, ArrayList<Integer> bannerImage) {
        this.mContext = context;
        this.bannerImage = bannerImage;
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return (view == (View)object);
    }
}

File: app/src/main/java/com/example/webtoonapp/BannerAdapter.kt
package com.example.webtoonapp

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import com.example.webtoonapp.R
import com.bumptech.glide.Glide

class BannerAdapter(private val context: Context, private val sliderImage: Array<String>) :
    RecyclerView.Adapter<BannerAdapter.MyViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.banner_viewpager_item, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val index = position % sliderImage.size
        holder.bindSliderImage(sliderImage[index])
        //        if (position % 2 == 0) {
//            holder.bindSliderImage(sliderImage[index]);
//        } else {
//            holder.bindSliderImage(sliderImage[index]);
//        }
    }

    override fun getItemCount(): Int {
        return Int.MAX_VALUE
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val mImageView: ImageView
        fun bindSliderImage(imageURL: String?) {
            Glide.with(context)
                .load(imageURL)
                .into(mImageView)
        }

        init {
            mImageView = itemView.findViewById(R.id.iv_bannerItem)
        }
    }
}